src/Adapter/Product/Image/Repository/ProductImageMultiShopRepository.php

```diff
@@ -0,0 +1,242 @@
+    {
+        $qb = $this->connection->createQueryBuilder()->select('id_image');
+
+        //todo: manage position
+        if ($shopConstraint->getShopGroupId()) {
+            throw new InvalidShopConstraintException('Image has no features related with shop group use single shop and all shops constraints');
+        } elseif ($shopConstraint->forAllShops()) {
+            $qb->from($this->dbPrefix . 'image', 'i')
+                ->addOrderBy('i.position', 'ASC')
+            ;
+        } else {
+            $qb->from($this->dbPrefix . 'image_shop', 'i')
+                ->andWhere('i.id_shop = :shopId')
+                ->setParameter('shopId', $shopConstraint->getShopId()->getValue())
+            ;
+        }
+
+        $results = $qb->andWhere('i.id_product = :productId')
+            ->setParameter('productId', $productId->getValue())
+            ->addOrderBy('i.id_image', 'ASC')
+            ->execute()
+            ->fetchAll()
+        ;
+
+        if (!$results) {
+            return [];
+        }
+
+        $imagesIds = [];
+        foreach ($results as $result) {
+            $imagesIds[] = new ImageId((int) $result['id_image']);
+        }
+
+        return $imagesIds;
+    }
+
+    /**
+     * @param ImageId $imageId
+     *
+     * @return Image
+     *
+     * @throws CoreException
+     */
+    public function get(ImageId $imageId, ShopId $shopId): Image
+    {
+        /** @var Image $image */
+        $image = $this->getObjectModelForShop(
+            $imageId->getValue(),
+            Image::class,
+            ProductImageNotFoundException::class,
+            $shopId
+        );
+
+        return $image;
+    }
+
+    /**
+     * @param ImageId $imageId
+     *
+     * @return ShopId[]
+     */
+    public function getAssociatedShopIds(ImageId $imageId): array
+    {
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->select('id_shop')
+            ->from($this->dbPrefix . 'image_shop')
+            ->where('id_image = :imageId')
+            ->setParameter('imageId', $imageId->getValue())
+        ;
+
+        $result = $qb->execute()->fetchAll();
+        if (empty($result)) {
+            return [];
+        }
+
+        $shops = [];
+        foreach ($result as $shop) {
+            $shops[] = new ShopId((int) $shop['id_shop']);
+        }
+
+        return $shops;
+    }
+
+    public function create(ProductId $productId, ShopConstraint $shopConstraint): Image
+    {
+        $productIdValue = $productId->getValue();
+        $image = new Image();
+        $image->id_product = $productIdValue;
+        $image->cover = !Image::getCover($productIdValue);
+
+        $shopIds = $this->productMultiShopRepository->getShopIdsByConstraint($productId, $shopConstraint);
+        $this->addObjectModelToShops($image, $shopIds, CannotAddProductImageException::class);
+
+        //Todo: Maybe useless try test behat with this
```

Comment:
Did you try if this is useless? Especially when creating a first image which is supposed to be the cover I believe `associateTo` doesn't handle this cover field at all.

Reply:
Yes next commit